sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -625,7 +625,8 @@ case class DescribeTableCommand(
         throw new AnalysisException(
           s"DESC PARTITION is not allowed on a temporary view: ${table.identifier}")
       }
-      describeSchema(catalog.lookupRelation(table).schema, result, header = false)
+      val schema = catalog.getTempViewOrPermanentTableMetadata(table).schema
```

Comment:
the view plan can be unresolved (with cast and alias added), we should use the recorded view schema.